File: models/Invoice.js
const mongoose = require("mongoose");

const invoiceSchema = new mongoose.Schema(
  {
    appointmentId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Appointment",
      required: true,
    },
    price: {
      type: Number,
      required: true,
    },
    status: {
      type: String,
      enum: ["paid", "unpaid", "cancelled","success"],
      default: "unpaid",
    },
    transactionId: {
      type: String,
      required: true,
    },
    metaData: {
      type: Object,
      required: false,
    },
    paymentType: {
      type: String,
      enum: ["chapa", "telebirr"],
      required: true,
    },
  },
  {
    timestamps: true,
    versionKey: false,
  }
);

module.exports = mongoose.model("Invoice", invoiceSchema);

File: lib/model/Invoice.dart
class Invoice {
  final String? id;
  final String? appointmentId;
  final String? price;
  final String? status;
  final String? paymentType;

  Invoice({this.id, this.appointmentId, this.price, this.status, this.paymentType});

  factory Invoice.fromJson(Map<String, dynamic> json) {
    return Invoice(
      id: json['_id'],
      appointmentId: json['appointmentId'],
      price: json['price'],
      status: json['status'],
      paymentType: json['paymentType'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'appointmentId': appointmentId,
      'price': price,
      'status': status,
      'paymentType': paymentType,
    };
  }
}
